modules/systests/systests.c

```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <tilck/kernel/debug_utils.h>
+#include <tilck/common/basic_defs.h>
+#include <tilck/kernel/elf_utils.h>
+#include <tilck/kernel/syscalls.h>
+#include <tilck/kernel/modules.h>
+#include <tilck/common/printk.h>
+#include <tilck/kernel/errno.h>
+#include <tilck/kernel/sched.h>
+
+#define TEST_VAR_VALUE 3
+
+long test_on_exit_cb_counter = TEST_VAR_VALUE;
+
+static void
+on_exit_test_callback(struct task *ti);
+
+void register_test_on_exit_callback(void)
+{
+   register_on_task_exit_cb(&on_exit_test_callback);
+}
+
+void unregister_test_on_exit_callback(void)
+{
+   unregister_on_task_exit_cb(&on_exit_test_callback);
+   test_on_exit_cb_counter = TEST_VAR_VALUE;
+}
+
+static void
+on_exit_test_callback(struct task *ti)
+{
+   test_on_exit_cb_counter += 1;
+}
+
+static int
+tilck_call_fn_0(const char *fn_name)
+{
+   const ulong fn_addr = find_addr_of_symbol(fn_name);
+
+   if (!fn_addr)
+      return -ENOENT;
+
+   ((void (*)(void))fn_addr)();
+
+   return 0;
+}
+
+static int
+tilck_get_var_long(const char *var_name, long *buf)
+{
+   const ulong var_addr = find_addr_of_symbol(var_name);
+
+   if (!var_addr) {
+      printk("Global variable %s not found\n", var_name);
+      return 1;
```

Comment:
Here you should return -ENOENT as in the other case, in order to the handling for the no-symbols case in the test to work.